### PR TITLE
chore:  rpk config field

### PR DESCRIPTION
**Pull request by alejandroEsc**
Fixes: #394 

Please check if the location of the new config makes sense. If it doesnt any suggestions welcome.

### CODE REVIEW

File: charts/redpanda/templates/configmap.yaml [hunk 1]
```diff
@@ -429,6 +429,23 @@ data:
 {{- /* END LISTENERS */}}
 
     rpk:
+{{- with (dig "rpk" dict .Values.config) }}
```
Comment (alejandroEsc) [inline charts/redpanda/templates/configmap.yaml:432]: This seems overly complicated, we could simplify if we were not concerned with configs being empty/null etc.. let me know if rpk configs are not as concerned as we are with redpanda.

Comment (RafalKorepta) [reply charts/redpanda/templates/configmap.yaml:432]: They should not be as much concerned as Redpanda config fileds.

**Comment by RafalKorepta** [comment]
https://github.com/redpanda-data/helm-charts/issues/394#issuecomment-1604129160 I'm not sure if this should be even merged.

**Comment by alejandroEsc** [comment]
from a values file entry of:
```
config:
  rpk:
    additional_start_flags:
      - "--idle-poll-time-us=0"
      - "--foo=bar"
    something: else
    blahblah: true
    additionally:
      blahblha: true
      something: no
```

we get in the configmap:

```
    rpk:
      additional_start_flags:
      - --idle-poll-time-us=0
      - --foo=bar
      additionally:
        blahblha: true
        something: false
      blahblah: true
      something: else
```

**Review by RafalKorepta — APPROVED** [review]

**Pull request merged**